piker/data/_web_bs.py

```diff
@@ -81,9 +83,15 @@ async def _connect(
                     trio_websocket.open_websocket_url(self.url)
                 )
                 # rerun user code fixture
-                ret = await self._stack.enter_async_context(
-                    self.fixture(self)
-                )
+                if self.token == '':
```

Comment:
why not just always pass the token in?
i would presume all web backends will need it.

Reply:
you could pass all (future) data like this as a `fixture_kwargs: dict` and then maybe just call it with `self.fixture(self, **self.fixture_kwargs)` and i think as long as the fixture doesn't declare it as a default kwarg it should just work?

Reply:
will be address in a new PR, see here #290